angular/projects/admin-nrpti/src/app/communications/lng-map-info/lng-map-info.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Constants } from '../../utils/constants/misc';
+import { MapInfo } from './../../../../../common/src/app/models/master/common-models/map-info';
+import { Router } from '@angular/router';
+@Component({
+  selector: 'app-lng-map-info',
+  templateUrl: './lng-map-info.component.html',
+  styleUrls: ['./lng-map-info.component.scss']
+})
+export class LngMapInfoComponent implements OnInit {
+  @Input() lngMapData: MapInfo[] = [];
+
+  public loading = true;
+  public mapForm: FormGroup;
+  public currentSection = null;
+
+  public sections = Constants.LngSectionPickList;
+
+  public tinyMceSettings = {
+    base_url: '/tinymce',
+    suffix: '.min',
+    browser_spellcheck: true,
+    height: 240,
+    plugins: ['lists, advlist, link'],
+    toolbar: [ 'undo redo | formatselect | ' +
+    ' bold italic backcolor | alignleft aligncenter ' +
+    ' alignright alignjustify | bullist numlist outdent indent |' +
+    ' removeformat | help' ]
+  };
+
+  constructor(
+    public router: Router,
+    private factoryService: FactoryService,
+    private loadingScreenService: LoadingScreenService,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) { }
+
+  ngOnInit() {
+    this.currentSection = this.lngMapData[0];
+    this.buildForm();
+  }
+
+  checkRoles() {
+    if (this.factoryService.userInRole('sysadmin')) {
+      return true;
+    } else if (this.factoryService.userInLngRole()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * When selecting a segment, update the form with it's corresponding data
+   *
+   * @param {object} event
+   * @returns
+   * @memberof LngMapInfoComponent
+   */
+  onSelectSegment(event) {
+    if (!event) {
+      return;
+    }
+    const segment = event.target.value.split(': ');
+    this.currentSection = this.lngMapData.find(elem => elem.segment === segment[1]);
+    this.buildForm();
+    this._changeDetectionRef.detectChanges();
+  }
+
+  private buildForm() {
+    this.mapForm = new FormGroup({
+      sectionNumber: new FormControl({
+        value: (this.currentSection && this.currentSection.segment) || '',
+        disabled: !this.checkRoles()
+      }),
+      location: new FormControl({
+        value: (this.currentSection && this.currentSection.location) || '',
+        disabled: !this.checkRoles()
+      }),
+      length: new FormControl({
+        value: (this.currentSection && this.currentSection.length) || '',
+        disabled: !this.checkRoles()
+      }),
+      recentUpdates: new FormControl({
+        value: (this.currentSection && this.currentSection.description) || '',
+        disabled: !this.checkRoles()
+      })
+    });
+  }
+
+  async submit() {
+    this.loadingScreenService.setLoadingState(true, 'main');
+
+    if (
+      !this.mapForm.get('location').value ||
+      !this.mapForm.get('length').value ||
+      !this.mapForm.get('recentUpdates').value
+    ) {
+      alert('Please ensure your updates to the Section info are complete');
```

Comment:
Should the code return here?  Currently it'll alert, but continue to save the empty values anyways.

How does the user "cancel" an update if `recentUpdates` can't be empty?

Reply:
ahh yes it should, I'll update this